src/coreclr/jit/simd.h

```diff
@@ -1993,6 +1993,104 @@ SveMaskPattern EvaluateSimdMaskToPattern(var_types baseType, simdmask_t arg0)
         }
     }
 }
+
+template <typename TSimd, typename TBase>
+SveMaskPattern EvaluateSimdVectorToPattern(TSimd arg0)
+{
+    uint32_t count    = sizeof(TSimd) / sizeof(TBase);
+    uint32_t finalOne = count;
+
+    // A mask pattern starts with zero or more 1s and then the rest of the mask is filled with 0s.
+    // This pattern is extracted using the least significant bits of the vector elements.
+
+    // Find an unbroken sequence of 1s.
+    for (uint32_t i = 0; i < count; i++)
+    {
+        TBase input0;
+        memcpy(&input0, &arg0.u8[i * sizeof(TBase)], sizeof(TBase));
+
+        // For Arm64 we have count total bits to read, but
+        // they are sizeof(TBase) bits apart. We set
+        // depending on if the corresponding input element
+        // has its least significant bit set
+
+        bool isSet = static_cast<uint64_t>(1) << (i * sizeof(TBase));
```

Comment:
as I pointed in other PR, this one doesn't even check with contents of `input0`.